torchtitan/checkpoint.py

```diff
@@ -20,6 +20,8 @@
     set_model_state_dict,
     set_optimizer_state_dict,
 )
+from torch.utils.data import DataLoader
```

Comment:
Why do we need this too?

Reply:
@andrewkho For typing. I can switch to StatefulDataLoader if we know that is going to be the one used in all scenarios